diff --git a/node_modules/nock/lib/request_overrider.js b/node_modules/nock/lib/request_overrider.js
index 4e51125..32be9f7 100644
--- a/node_modules/nock/lib/request_overrider.js
+++ b/node_modules/nock/lib/request_overrider.js
@@ -151,6 +151,14 @@ function RequestOverrider(req, options, interceptors, remove, cb) {
   };
 
   req.end = function(buffer, encoding, callback) {
+    if (typeof buffer === 'function') {
+      buffer = undefined;
+      encoding = undefined;
+      callback = buffer;
+    } else if (typeof encoding === 'function') {
+      encoding = undefined;
+      callback = encoding;
+    }
     debug('req.end');
     if (!req.aborted && !ended) {
       req.write(buffer, encoding, function () {
